pkg/controller/bundles/bundles_syncer.go

```diff
@@ -48,40 +74,78 @@ func (syncer *LeafHubBundlesSpecSync) Start(stopChannel <-chan struct{}) error {
 	ctx, cancelContext := context.WithCancel(context.Background())
 	defer cancelContext()
 
-	go syncer.sync(ctx)
+	// start workers
+	for i := 0; i < len(syncer.k8sClients); i++ {
+		go syncer.runClientWorker(ctx, syncer.k8sClients[i])
+	}
 
-	for {
-		<-stopChannel // blocking wait for stop event
-		syncer.log.Info("stopped bundles syncer")
-		cancelContext()
+	go syncer.sync()
 
-		return nil
-	}
+	<-stopChannel // blocking wait for stop event
+	syncer.log.Info("stopped bundles syncer")
+
+	return nil
 }
 
-func (syncer *LeafHubBundlesSpecSync) sync(ctx context.Context) {
+func (syncer *LeafHubBundlesSpecSync) sync() {
 	syncer.log.Info("start bundles syncing...")
 
 	for {
```

Comment:
(addressed in the comment above) 

Should handle cancellation of the received context, I see it is missing in the original code.

Reply:
done